### PR TITLE
Add nox configuration to start pyodine console

### LINKED ISSUE #3330: Create a workflow (nox?) for testing Emscripten support locally
Would be great to have an easy-to-use workflow for contributors to:

* Run a single command
* Have all dependencies installed
* Opens a Pyodide console in a web browser
* Makes the local copy of urllib3 available for installation with micropip/Pyodide

This would help greatly with being able to poke around with and develop Emscripten/Pyodide support.

### PR DESCRIPTION
**Pull request by ecerulm**
This adds a nox session called `pyodineconsole`. 

```
nox -rs pyodineconsole
nox > Running session pyodineconsole
nox > Re-using existing virtual environment at .nox/pyodineconsole.
nox > python -m pip install build
nox > python -m build
* Creating venv isolated environment...
* Installing packages in isolated environment... (hatchling>=1.6.0,<2)
* Getting build dependencies for sdist...
* Building sdist...
* Building wheel from sdist
* Creating venv isolated environment...
* Installing packages in isolated environment... (hatchling>=1.6.0,<2)
* Getting build dependencies for wheel...
* Building wheel...
Successfully built urllib3-2.2.0.tar.gz and urllib3-2.2.0-py3-none-any.whl
nox > cp templates/pyodide-console.html dist/index.html
nox > cd dist
nox > python -m http.server
Serving HTTP on :: port 8000 (http://[::]:8000/) ...
```

This nox session
* builds the wheel for `urllib` into `dist/`
* creates a `index.html` file in `dist/`, with a pyodine terminal taken 
from [pyodine src/templates/console.html](https://github.com/pyodide/pyodide/blob/main/src/templates/console.html)
* autoruns a series of commands on this terminal 
  * `import micropip`
  * `await micropip.install("http://localhost:8000/urllib3-2.2.0-py3-none-any.whl")`
  * `import urllib3`
  * `urllib3.__version__`

Then you open http://localhost:8000 and you will be presented with 

image

Closes #3330

### CODE REVIEW

**Comment by ecerulm** [comment]
@sethmlarson , please try this pyodide console approach and tell me if that's what you were looking for when you wrote #3330

File: noxfile.py [hunk 1]
```diff
@@ -200,6 +200,18 @@ def lint(session: nox.Session) -> None:
     mypy(session)
 
 
+@nox.session(python="3.11")
```
Comment (sethmlarson) [inline noxfile.py:203]: Does this need to be a specific Python version, or can we use any 3.x?

File: noxfile.py [hunk 2]
```diff
@@ -200,6 +200,18 @@ def lint(session: nox.Session) -> None:
...
+def pyodineconsole(session: nox.Session) -> None:
+    # build wheel into dist folder
+    session.install("build")
+    session.run("python", "-m", "build")
+    session.run(
+        "cp", "templates/pyodide-console.html", "dist/index.html", external=True
+    )
+    session.cd("dist")
+    session.run("python", "-m", "http.server")
```
Comment (sethmlarson) [inline noxfile.py:212]: Let's specify a port here and then open the browser automatically, I believe there's a Python function for that?

File: templates/pyodide-console.html [hunk 3]
```diff
@@ -0,0 +1,267 @@
+<!doctype html>
```
Comment (sethmlarson) [inline templates/pyodide-console.html:1]: Can we find a new home for this file? Maybe under `test/` for Emscripten?

Did you write this file yourself or is it from somewhere, I don't see license information.

What is the expected maintenance requirements for this file?

**Review by sethmlarson — COMMENTED** [review]
Thanks for opening this! I put some early comments below:

File: noxfile.py (hunk 2 above)
Comment (ecerulm) [reply noxfile.py:212]: but I don't think we can let the `python -m http.server` in the background with nox. So we can open the browser before the server is up , but that would be great either.

Comment (sethmlarson) [reply noxfile.py:212]: Oh right, yeah let's keep this as-is!

File: noxfile.py (hunk 1 above)
Comment (ecerulm) [reply noxfile.py:203]: If we don't specify a version then nox will run the same python interpreter used to run nox itself. 

I mean, the alternative skip nox (which is for test automation, not for this). 

I can create a python script to create virtualenv , install dependencies, etc instead. then we are not bound by the nox rules.

File: noxfile.py (hunk 2 above)
Comment (ecerulm) [reply noxfile.py:212]: I can create a python script to create virtualenv , install dependencies, etc instead. then we are not bound by the nox rules.

File: templates/pyodide-console.html [hunk 4]
```diff
@@ -0,0 +1,267 @@
...
+        if (searchParams.has("noblink")) {
+          $(".cmd-cursor").addClass("noblink");
+        }
+        await term.ready;
+        await term.exec("import micropip\n");
+        await term.exec("micropip.list()\n");
+        await term.exec('await micropip.install("http://localhost:8000/urllib3-2.2.0-py3-none-any.whl")')
+        await term.exec("micropip.list()");
+        await term.exec("import urllib3");
+        await term.exec("urllib3.__version__");
```
Comment (ecerulm) [inline templates/pyodide-console.html:262]: @sethmlarson , this is the part that I added to the original https://github.com/pyodide/pyodide/blob/3500c17468e474902044413679b024dc7af75621/src/templates/console.html

File: templates/pyodide-console.html (hunk 3 above)
Comment (ecerulm) [reply templates/pyodide-console.html:1]: > Can we find a new home for this file? Maybe under test/ for Emscripten?

Sure, I can do that. 


> Did you write this file yourself or is it from somewhere, I don't see license information.

I mentioned in the PR description 
>> creates a index.html file in dist/, with a pyodine terminal taken
from [pyodide src/templates/console.html]

The LICENSE is https://github.com/pyodide/pyodide/blob/main/LICENSE. (Mozilla Public License Version 2.0) 

I can add a comment on the top of the file with the original source url, not sure , should I add a "copyright" comment although the original does not have it? The closest copyright notice that I can find in their repo is  "2019-2022, Pyodide contributors and Mozilla", I can add that?  


> What is the expected maintenance requirements for this file?

I guess we won't maintain this file. It's the same jQuery REPL that is used in https://pyodide.org/en/stable/console.html, if it stops working we can always copy the latest from [pyodide src/templates/console.html]. And then add the lines

```
        await term.ready;
        await term.exec("import micropip\n");
        await term.exec("micropip.list()\n");
        await term.exec('await micropip.install("http://localhost:8000/urllib3-2.2.0-py3-none-any.whl")')
        await term.exec("micropip.list()");
        await term.exec("import urllib3");
        await term.exec("urllib3.__version__");
```

 [pyodide src/templates/console.html]: https://github.com/pyodide/pyodide/blob/main/src/templates/console.html

Comment (sethmlarson) [reply templates/pyodide-console.html]: Thank you, apologies for not reading the description closely enough

**Review by sethmlarson — APPROVED** [review]
This is awesome and perfectly recreates the issue w/ InsecureRequestWarning, thank you @ecerulm!

**Comment by sethmlarson** [comment]
@ecerulm The linked issue was a bounty issue for $100, so if you're able to please submit an expense to our Open Collective linking back to this PR.

**Pull request merged**